### PR TITLE
Gt 155 uart name packed string interfaceitem

### PR DESCRIPTION
**Pull request by epcAnker**
https://github.com/epcpower/grid-tied/pull/58

### CODE REVIEW

**Comment by gordon-epc** [comment]
Right off the bat, I can see you will need to reformat the file using black.

File: src/epcpm/parameterstointerface.py [hunk 1]
```diff
@@ -303,21 +305,25 @@ def interface_variable_name(self):
         model = maybe_model
         model_variable = f"sunspecInterface.model{model.id}"
 
-        return f"&{model_variable}.{parameter.abbreviation}"
-
+        if reference == True:
+            return f"&{model_variable}.{parameter.abbreviation}"
+        else:
+            return f"{model_variable}.{parameter.abbreviation}"
```
Comment (gordon-epc) [inline src/epcpm/parameterstointerface.py:311]: Consider replacing with arguably not easier to read one liner:
`return f"{'&' if reference else ''}{model_variable}.{parameter.abbreviation}"`

File: src/epcpm/parameterstointerface.py [hunk 2]
```diff
@@ -303,21 +305,25 @@ def interface_variable_name(self):
...
+    def interface_variable_name(self, reference=True):
         parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
 
         uuid_ = str(parameter.uuid).replace("-", "_")
-        return f"&interfaceItem_variable_{uuid_}"
-
+        if reference == True:
+            return f"&interfaceItem_variable_{uuid_}"
+        else:
+            return f"interfaceItem_variable_{uuid_}"
```
Comment (gordon-epc) [inline src/epcpm/parameterstointerface.py:326]: Consider replacing with arguably not easier to read one liner:
`return f"{'&' if reference else ''}interfaceItem_variable_{uuid_}"`

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: src/epcpm/parameterstointerface.py [hunk 3]
```diff
@@ -1410,6 +1428,11 @@ def create_item(
         include_uuid_in_item=include_uuid_in_item,
     )
 
+    maybe_sunspec_variable_length = []
+    if parameter.internal_type == "UartName":
```
Comment (altendky) [inline src/epcpm/parameterstointerface.py:1432]: I think (maybe) this function so far doesn't have any significant type awareness?  Perhaps this goes outside and comes in more like `variable_or_getter_setter`?

**Review by altendky — CHANGES_REQUESTED** [review]

File: src/epcpm/parameterstointerface.py [hunk 4]
```diff
@@ -292,6 +293,7 @@ class DataPoint:
     wrapped = attr.ib()
     parameter_uuid_finder = attr.ib()
 
+    # By Default will return reference to the variable unless reference parameter == false.
```
Comment (altendky) [inline src/epcpm/parameterstointerface.py:296]:
```suggestion
```

File: src/epcpm/parameterstointerface.py [hunk 5]
```diff
@@ -316,7 +318,8 @@ def interface_variable_name(self):
         parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
 
         uuid_ = str(parameter.uuid).replace("-", "_")
-        return f"&interfaceItem_variable_{uuid_}"
+
```
Comment (altendky) [inline src/epcpm/parameterstointerface.py:321]: Just to reduce diff noise.  `*shrug*`
```suggestion
```

File: src/epcpm/parameterstointerface.py [hunk 6]
```diff
@@ -1472,15 +1494,22 @@ def create_common_initializers(
         "true" if reject_from_inactive_interfaces else "false"
     )
 
+    scale_factor_variable_ref = "NULL"
+    if scale_factor_variable is not "NULL":
```
Comment (altendky) [inline src/epcpm/parameterstointerface.py:1498]: Short strings tend to be interned and thus `is` works, but it is not guaranteed.

```suggestion
    if scale_factor_variable != "NULL":
```

File: src/epcpm/parameterstointerface.py [hunk 7]
```diff
@@ -1472,15 +1494,22 @@ def create_common_initializers(
...
+        scale_factor_variable_ref = "&" + scale_factor_variable
+    sunspec_variable_ref = "NULL"
+    if sunspec_variable is not "NULL":
```
Comment (altendky) [inline src/epcpm/parameterstointerface.py:1501]:
```suggestion
    if sunspec_variable != "NULL":
```

**Review by altendky — CHANGES_REQUESTED** [review]
Just small stuff this time. `:]`

File: src/epcpm/parameterstointerface.py (hunk 4 above)
Comment (epcAnker) [reply src/epcpm/parameterstointerface.py:296]: good catch

**Review by gordon-epc — APPROVED** [review]
Run black and it should be all good to go.

**Pull request merged**